src/rgw/rgw_lc.h

```diff
@@ -515,8 +515,11 @@ class RGWLC : public DoutPrefixProvider {
   void initialize(CephContext *_cct, rgw::sal::Store* _store);
   void finalize();
 
-  int process(LCWorker* worker, bool once);
-  int process(int index, int max_secs, LCWorker* worker, bool once);
+  int process(LCWorker* worker,
+	      std::unique_ptr<rgw::sal::Bucket>& optional_bucket, bool once);
```

Comment:
nit: pass `rgw::sal::Bucket` as a raw pointer if the function isn't participating in its lifetime management. that way callers can also pass a pointer to variables on the stack, stored in a shared_ptr, etc. passing a mutable reference to unique_ptr also implies that the function may reassign or reset the unique_ptr

Reply:
I think, in our codebase, the fact that this pointer is by default managed with a unique_ptr provides better guidance on the semantics.  Passing the reference allowed optional semantics for free, which I specifically wanted.

Reply:
raw pointers have the same optional semantic. but if you want to keep unique_ptr here, please pass by const reference